src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -678,6 +678,101 @@ bool QgsMapToolLabel::currentLabelDataDefinedPosition( double &x, bool &xSuccess
   return true;
 }
 
+bool QgsMapToolLabel::currentLabelDataDefinedPosition( double &x, bool &xSuccess, double &y, bool &ySuccess, int &xCol, int &yCol, int &pointCol ) const
```

Comment:
Can you extend the existing currentLabelDataDefinedPosition method instead of reimplementing it entirely? This is a LOT of duplicate code